prereqScraper.py

```diff
@@ -0,0 +1,174 @@
+
+    print("}")
+    return
+
+
+def formatLink(link):
+    link = link.strip(" ")
+    link = link.strip(".")
+    link = link.strip(",")
+    return link
+
+
+courses = pd.DataFrame(columns=[
+    "course_code",
+    "course_name",
+    "antirequisites",
+    "prerequisites",
+    "description",
+    "location",
+    "extra_info"])
+
+for cat in catList[43:]:
+    print(cat)
+    if SKIP_SCRAPE:
+        break
+    url = f"https://www.westerncalendar.uwo.ca/Courses.cfm?Subject={cat}&SelectedCalendar=Live&ArchiveID="
+
+    page = requests.get(url)
+    soup = BeautifulSoup(page.text, "html.parser")
+    for panel in soup.find_all(class_="panel-default")[2:]:
+        prereqs = []  # [string, link?, bold?]
+        antireqs = []
+        coreqs = []
+        preAndCoreqs = []
+        prereqsLink = []  # [string, link?, bold?]
+        antireqsLink = []
+        coreqsLink = []
+        preAndCoreqsLink = []
+        coreqFlag = False
+        pcreqFlag = False
+        title = panel.find_all(class_="courseTitleNoBlueLink")[
+            0].get_text().strip()
+        desc = panel.find_all(
+            class_="col-xs-12")[0].get_text().strip().replace("\n", "").replace("\r", "")
+        location = panel.find_all("img")[0]["alt"]
+        for child in panel.find_all(class_="col-xs-12"):
+            if child.find_all("strong"):
+                strongText = child.find_all("strong")[0].get_text()
+                if strongText == "Antirequisite(s):":
+                    for item in child.children:
+                        if hasattr(item, "children"):
+                            for req in item.children:
+                                reqTraits = [req.get_text().strip().replace(
+                                    "\n", "").replace("\r", ""), False]
+                                if not reqTraits[0]:
+                                    continue
+                                if type(req) == Tag:
+                                    if req.name == "a":
+                                        reqTraits[1] = True
+                                        antireqsLink.append(
+                                            formatLink(reqTraits[0]))
+                                antireqs.append(
+                                    {"text": reqTraits[0], "isLink": reqTraits[1]})
+                elif strongText == "Prerequisite(s):" or "Corequisite(s):" in strongText:
+                    for item in child.children:
+                        if hasattr(item, "children"):
+                            for req in item.children:
+                                reqTraits = [req.get_text().strip().replace(
+                                    "\n", "").replace("\r", ""), False]
+                                if not reqTraits[0]:
+                                    continue
+                                if "Prerequisite(s):" in reqTraits[0]:
+                                    continue
```

Comment:
Too many `continue`s for my brain LMAO... Could you clean up the logic here (like the above suggestion)? OR if you want to reduce the amount of nested if blocks, I'd be ok with one continue here.

```suggestion
                                if not reqTraits[0] and "Prerequisite(s):" in reqTraits[0]:
                                    continue
```